Token demo

ZiadAbdelkarim opened the pull request:
Creation of token demo and accompanying documentation.

Resolves #13.

ZiadAbdelkarim committed:
Created new Token passing demo along with tutorial description

+ Token demo tutorial section

ZiadAbdelkarim committed:
Created new Token passing demo along with tutorial description

+ Token demo tutorial section

ZiadAbdelkarim committed:
Modified Token Demo

+ State field and labelOfFirstNbrInState(...) in tokenDemo class
+ Tokens disappear after being passed x times.

ZiadAbdelkarim committed:
Modified Token Demo

+ State field and labelOfFirstNbrInState(...) in tokenDemo class
+ Tokens disappear after being passed x times.

ZiadAbdelkarim committed:
Completed the tokentutorial

ZiadAbdelkarim committed:
Completed the tokentutorial

jdaymude committed:
~ Updated TokenDemo to pass red and blue tokens in opposite directions

~ Rebased onto dev
- Removed unnecessary functions
~ Updated inconsistent in-line documentation
+ Added termination function

jdaymude commented:
@ZiadAbdelkarim I've significantly updated the structure of the demo code so that the behavior is easier to see (now, red and blue tokens are passed in opposite directions). I hadn't realized how confusing passing in random directions was going to look. Please take a stab at updating your documentation accordingly.

ZiadAbdelkarim committed:
Merge remote-tracking branch 'origin/token-demo' into token-tutorial

ZiadAbdelkarim committed:
Updated documentation to match implementation

ZiadAbdelkarim committed:
Merge branch 'dev' into token-tutorial

ZiadAbdelkarim committed:
~branch up to date with dev following new algorithm instantiation

jdaymude committed:
~ Revised TokenDemo tutorial

~ Updated TokenDemo to take a token lifetime parameter as input
+ Reorganized token functions in AmoebotParticle and added a missing overloaded version of countTokens
~ Minor formatting and documentation updates
- Removed duplicate ui/alg.cpp

jdaymude reviewed (approved):
LGTM
